tests/Feature/PostsOfUserApiTest.php

```diff
@@ -0,0 +1,190 @@
+        $response = $this->json('GET', 'api/users/' . $user->id . '/posts');
+        $response->assertStatus(Response::HTTP_OK);
+    }
+
+    /**
+     * Test struct json.
+     *
+     * @return void
+     */
+    public function testStructJson()
+    {
+        $user = User::first();
+        $this->be($user);
+        $response = $this->json('GET', 'api/users/' . $user->id . '/posts');
+        $response->assertJsonStructure([
+            'meta' => [
+                'status',
+                'code'
+            ],
+            'data' => [
+                [
+                    'id',
+                    'user_id',
+                    'content',
+                    'status',
+                    'name',
+                    'team',
+                    'avatar_url',
+                    'is_admin',
+                    'picture',
+                    'title',
+                    'book_id',
+                    'rating',
+                    'likes',
+                    'created_at',
+                    'updated_at',
+                ],
+            ],
+            'pagination' => [
+                'total',
+                'per_page',
+                'current_page',
+                'total_pages',
+                'links' => [
+                    'prev',
+                    'next'
+                ]
+            ],
+        ]);
+        $response->assertStatus(Response::HTTP_OK);
+    }
+
+    /**
+     * Test check some object compare database.
+     *
+     * @return void
+     */
+    public function testCompareDatabase()
+    {
+        $user = User::first();
+        $this->be($user);
+        $response = $this->json('GET', 'api/users/' . $user->id . '/posts');
+        $data = json_decode($response->getContent())->data;
+        foreach ($data as $post) {
+            $arrayCompare = [
+                'id' => $post->id,
+                'content' => $post->content,
+            ];
+            $this->assertDatabaseHas('posts', $arrayCompare);
+        }
+    }
+
+    /**
+     * Test result pagination.
+     *
+     * @return void
+     */
+    public function testGetPaginationResult()
+    {
+        $user = User::first();
+        $this->be($user);
+        $response = $this->json('GET', 'api/users/' . $user->id . '/posts?page=2');
+        $response->assertJson([
+            'pagination' => [
+                'total' => self::NUMBER_RECORD_CREATE,
+                'per_page' => config('define.posts.limit_rows_posts_of_user'),
+                'current_page' => 2,
+                'total_pages' => 2,
+            ]
+        ]);
+        $response->assertStatus(Response::HTTP_OK);
+    }
+
+    /**
+     * Test result pagination.
+     *
+     * @return void
+     */
+    public function testPaginationResultHasFilter()
+    {
+        $user = User::first();
+        $this->be($user);
+        $response = $this->json('GET', 'api/users/' . $user->id . '/posts?status=' . Post::TYPE_REVIEW_BOOK . '&page=2');
```

Comment:
tạo mảng 
$data = [
 'status' => Post::TYPE_REVIEW_BOOK,
 'page' => 2
]
 $response = $this->get('api/users/' . $user->id . '/posts , $data);
nhìn đỡ rối hơn